### PR TITLE
Add validation required fix and corresponding package tests

### PR DESCRIPTION
**Pull request by galaxie**
Function `Required` had, I suppose, not desired behaviour due to not correct usage of `reflect` package.
Also I added some tests to `validate` package because there were not that many of them.

### CODE REVIEW

File: httpkit/validate/values.go [hunk 1]
```diff
@@ -99,7 +99,7 @@ func MaxLength(path, in, data string, maxLength int64) *errors.Validation {
 // Required validates an interface for requiredness
 func Required(path, in string, data interface{}) *errors.Validation {
 	val := reflect.ValueOf(data)
-	if reflect.DeepEqual(reflect.Zero(val.Type()), val) {
+	if reflect.DeepEqual(reflect.Zero(val.Type()).Interface(), val.Interface()) {
```
Comment (casualjim) [inline httpkit/validate/values.go]: what happens when there is a nil passed into this?

File: httpkit/validate/values_test.go [hunk 2]
```diff
@@ -28,3 +28,81 @@ func TestValidateEnum(t *testing.T) {
...
+	err = MaxLength("test", "body", "aa", minLength)
+	assert.NoError(t, err)
+}
+
+func TestValidateRequired(t *testing.T) {
+	var err error
+	path := "test"
+	in := "body"
+	RequiredFail := []interface{}{
+		"",
```
Comment (casualjim) [inline httpkit/validate/values_test.go]: can 'nil' be added to this test?

Comment (galaxie) [inline httpkit/validate/values_test.go]: ok

**Comment by galaxie** [comment]
I am not that sure but aren't these CI ("to long") errors kinda common for the current branch? Am I supposed care about them in this pull request to make it accepted?

**Comment by casualjim** [comment]
lgtm
sorry I fixed those in anohter branch

**Pull request merged**